src/interaction/orbitalnavigator.cpp

```diff
@@ -1173,6 +1289,32 @@ glm::dvec3 OrbitalNavigator::translateHorizontally(double deltaTime,
     return cameraPosition + rotationDiffVec3;
 }
 
+glm::dvec3 OrbitalNavigator::moveCameraAlongVector(const glm::dvec3& camPos,
+                                                   double distFromCameraToFocus,
+                                                  const glm::dvec3& camPosToAnchorPosDiff,
+                                                                        double focusLimit,
+                                                                         bool flyTo) const
+{
+    double velocityFactor = 0;
+    const double focusLimitBuffer = 0.1;
+
+    // Calculate and truncate the factor which determines the velocity of the 
+    // camera movement towards the focus node
+    if (flyTo) {
+        velocityFactor = (1 - ((focusLimit * (1.0 - focusLimitBuffer))
+        / distFromCameraToFocus));
+    }
+    else {
+        velocityFactor = (1 - (distFromCameraToFocus
+        / (focusLimit * (1.0 + focusLimitBuffer))));
+    }
+    velocityFactor *= _velocitySensitivity;
+    velocityFactor = floor(velocityFactor * 1000) / 1000;
```

Comment:
Why is this necessary? Why 1000?

Reply:
This code has been updated now, simplyfying a lot of the math.